core/src/View/MDSpan/Kokkos_MDSpan_Header.hpp

```diff
@@ -44,7 +44,39 @@ using std::layout_stride;
 using std::mdspan;
 }  // namespace Kokkos
 #else
+// Opt in for Kokkos::pair to submdspan/subview
+// submdspan does only take index_pair_like which is derived from tuple_like
+// tuple_like is an enumerated list: tuple, pair, array, complex,
+// ranges::subrange Needs to be defined before including mdspan header
+
+#include <Kokkos_Pair.hpp>
+
+namespace Kokkos {
+namespace detail {
+template <class IdxT1, class IdxT2>
+KOKKOS_INLINE_FUNCTION constexpr auto first_of(
+    const pair<IdxT1, IdxT2> &slice) {
```

Comment:
ADL to the rescue. 🙂 I'd still prefer to be explicit about the proper namespace like
```suggestion
    const ::Kokkos::pair<IdxT1, IdxT2> &slice) {
```

Reply:
What do you mean with ADL? In line 54 we are opening the Kokkos namespace ...

Reply:
Nevermind. I just got confused seeing the unqualified `pair` here.